### ISSUE TITLE
Using sol-trace with external, existing contracts as interfaces 

### ISSUE THREAD

**Issue by fxfactorial** [0xProject/0x-monorepo#2679]
thank you for sol-trace 

I am trying to use it with external libraries - specifically from curve.fi 

so I have 

```
const projectRoot = '.';
const solcVersion = '0.6.5+commit.f956cc89';
const artifactAdapter = new TruffleArtifactAdapter(projectRoot, solcVersion);

const revertTraceSubprovider = new RevertTraceSubprovider(
  artifactAdapter,
  defaultFromAddress,
  isVerbose
);
const provider = new ProviderEngine();

provider.addProvider(revertTraceSubprovider);
provider.addProvider(
  new WebsocketSubprovider({rpcUrl: 'http://localhost:8545'})
);

provider.start((err) => {
  if (err !== undefined) {
    console.log(err);
    process.exit(1);
  }
});

provider.send = provider.sendAsync.bind(provider);

module.exports = {
  networks: {
    // each name is really a key that you can use
    // truffle console --network forked
    forked: {
      provider,
      // host: '127.0.0.1', // Localhost (default: none)
      // port: 8545, // Standard Ethereum port (default: none)
      // 3 is ropsten
      network_id: '1', // Any network (default: none)
      skipDryRun: true,
    },
}
```

and this works fine - the context is that I am running it against `ganache-cli --fork` and the node it is talking to is my own locally running openethereum node. 

the interface for curve I'm using is 

```
interface ICurveFi {
  function exchange_underlying(
    int128 from,
    int128 to,
    uint256 _from_amount,
    uint256 _min_to_amount
  ) external;
}
```

and when I try to invoke the tracer under code that explodes, under truffle test, I get 

```
Unable to find matching bytecode for contract address 0x45f783cce6b7ff23b2ab2d70e416cdb7d6055f51, please check your artifacts. Ignoring...
```

which does kinda make sense since I don't have the bytecode for this - its an interface - 

the code is reverting at where I suspected - but the stack trace is not that meaningful - it only shows it place where the revert happened - but doesn't go any deeper than that because i don't have the bytecode for these external libaries. 

I am not sure what the right flow is for this at this point - I tried a hack of `eth.getCode(..)` and adding it to the build .json respect of the interface i was using, adding it to the bytecode key of the json, but this is a huge hack because the bytecode was build under no idea version of solc/i don't have enough invariants on its build. 

TL;DR 

how to use sol-trace with external contracts that you only expose as interfaces?

**Comment by rori4** [comment]
I have the exact same issue. Any workaround on this?
